data/core/common.lua

```diff
@@ -322,4 +322,46 @@ function common.mkdirp(path)
   return true
 end
 
+function common.rmdir(path, recursively)
+  local stat = system.get_file_info(path)
+  if not stat or stat.type ~= "dir" then
+    return false, "given path is not a directory", path
+  end
+
+  local contents = system.list_dir(path)
+  if #contents > 0 and not recursively then
+    return false, "directory is not empty", path
+  end
+
+  if #contents > 0 then
```

Comment:
Isn't this `if` redundant with the `for` loop over contents that just follows ?

Reply:
That was to not execute the loop in case that the directory was empty but it also works fine without it so removed it.